Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java

```diff
@@ -3602,11 +3609,87 @@ private LatLng getNewLatLng(float fraction, LatLng a, LatLng b) {
         return new LatLng(lat, lng);
     }
 
+
+    private  void animatePolyline(final int staticColor, final int animateColor){
+        polylineAnimatorSet =  new AnimatorSet();
+
+        polylineDrawingAnimator = ValueAnimator.ofInt(0, 100);
+        polylineDrawingAnimator.setDuration(700);
+        polylineDrawingAnimator.setInterpolator(new AccelerateDecelerateInterpolator());
+
+        polylineDrawingAnimator.addUpdateListener(animation -> {
+            if(polylines != null){
+                List<LatLng> foregroundPoints = polylines.getPoints();
+                Collections.reverse(foregroundPoints);
+                int percentageValue = (int) animation.getAnimatedValue();
+                int pointCount = foregroundPoints.size();
+                int countToBeRemoved = (int) (pointCount * (percentageValue / 100.0f));
+                List<LatLng> subListToBeRemoved = foregroundPoints.subList(0, countToBeRemoved);
+                subListToBeRemoved.clear();
+                polylines.setColor(animateColor);
+                overlayPolylines.setColor(staticColor);
+                overlayPolylines.setPoints(foregroundPoints);
+            }
+        });
+        polylineDrawingAnimator.addListener(new Animator.AnimatorListener() {
```

Comment:
The polylineDrawingAnimator have empty methods. If you don't require any specific behavior in these methods, you can remove them from the code.

Reply:
done